uberfire-widgets/uberfire-widgets-commons/src/main/java/org/uberfire/ext/widgets/common/client/common/DatePicker.java

```diff
@@ -238,12 +250,75 @@ public void setForceParse( final boolean forceParse ) {
      * Set the format of the Date shown in the TextBox component.
      * This is converted to BS3's Date Format that the underlying jQuery-based BS3 DatePicker
      * uses to convert values in the TextBox to selections in the popup date picker element.
-     * @param gwtDateFormat
+     * @param dateFormat
      */
-    public void setFormat( final String gwtDateFormat ) {
-        this.gwtDateFormat = gwtDateFormat;
+    public void setFormat( final String dateFormat ) {
+        setFormat(LocaleInfo.getCurrentLocale().getLocaleName(),dateFormat);
+    }
+
+    public void setFormat(String currentLang,  final String dateFormat) {
+        this.gwtDateFormat = setDatePickerLangProperties(currentLang, dateFormat);
         this.gwtDateTimeFormat = DateTimeFormat.getFormat( this.gwtDateFormat );
-        datePicker.setFormat( DatePickerFormatUtilities.convertToBS3DateFormat( gwtDateFormat ) );
+        datePicker.setFormat(DatePickerFormatUtilities.convertToBS3DateFormat(this.gwtDateFormat));
+    }
+
+    /**
+     * Sets the org.gwtbootstrap3.extras.datepicker.client.ui.DatePicker language and week start day
+     * depending on the activeLocale passed as parameter.
+     * There are incompatibilities between DateTimeFormatInfoImpl and DatePickerResourceBundles for es and de locales.
+     * This function modify in that cases the dateFormat to avoid show the name of the months
+     * TODO review locales es and de showing the name of the month.
+     * @param activeLocale
+     * @param dateFormat
+     * @return the dateformat to be applied depending of locale. In case of es and de, avoiding show the name of month
+     */
+    private String setDatePickerLangProperties(String activeLocale, String dateFormat){
+        DatePickerLanguage datePickerLanguage = DatePickerLanguage.EN;
+        DatePickerDayOfWeek weekStart = DatePickerDayOfWeek.SUNDAY;
+        String format = dateFormat;
+        if (activeLocale.equals("es")) {
+            datePickerLanguage = DatePickerLanguage.ES;
```

Comment:
Personally, I'd try to find another way of doing the conversion from ```activeLocale``` to ```DatePickerLanguage```.. if another locale is added for BxMS we'd need to remember to change this class. Did you consider ```DatePickerLanguage.valueOf(..)``` with some minimal manipulation of ```activeLocale```?